File: 151002/Rusakovich/NewsDistributedSystem/publisher/src/main/java/by/rusakovich/publisher/service/impl/LabelService.java
package by.rusakovich.publisher.service.impl;

import by.rusakovich.publisher.dao.IEntityRepository;
import by.rusakovich.publisher.model.dto.label.LabelRequestTO;
import by.rusakovich.publisher.model.dto.label.LabelResponseTO;
import by.rusakovich.publisher.model.dto.mapper.EntityMapper;
import by.rusakovich.publisher.model.entity.impl.jpa.JpaLabel;
import by.rusakovich.publisher.service.EntityService;
import org.springframework.stereotype.Service;

@Service
public class LabelService extends EntityService<Long, LabelRequestTO, LabelResponseTO, JpaLabel> {
    public LabelService(EntityMapper<Long, JpaLabel, LabelRequestTO, LabelResponseTO> mapper, IEntityRepository<Long, JpaLabel> rep) {
        super(mapper, rep);
    }
}

File: 151001/Hlebanova/LR4/Publisher/Service/Implementation/LabelService.cs
using AutoMapper;
using Publisher.Entity.Db;
using Publisher.Entity.DTO.RequestTO;
using Publisher.Entity.DTO.ResponseTO;
using Publisher.Repository.Interface;
using Publisher.Service.Implementation.Common;
using Publisher.Service.Interface;

namespace Publisher.Service.Implementation
{
    public class LabelService(IMapper mapper, ILabelRepository repository)
        : AbstractCrudService<Label, LabelRequestTO, LabelResponseTO>(mapper, repository), ILabelService
    {
        public override async Task<LabelResponseTO> Add(LabelRequestTO label)
        {
            if (!Validate(label))
            {
                throw new InvalidDataException("LABEL is not valid");
            }

            return await base.Add(label);
        }

        public override async Task<LabelResponseTO> Update(LabelRequestTO label)
        {
            if (!Validate(label))
            {
                throw new InvalidDataException($"UPDATE invalid data: {label}");
            }

            return await base.Update(label);
        }

        public Task<IList<LabelResponseTO>> GetByIssueID(int issueId)
        {
            throw new NotImplementedException();
        }

        private static bool Validate(LabelRequestTO label)
        {
            var nameLen = label.Name.Length;

            if (nameLen < 2 || nameLen > 32)
            {
                return false;
            }
            return true;
        }
    }
}
